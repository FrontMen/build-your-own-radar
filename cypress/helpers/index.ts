export const dataCy = (value: string) => cy.get(`[data-cy=${value}]`);
export const dataTestId = (value: string) => cy.get(`[data-testid=${value}]`);

export const hexToRgb = (hex: string) => {
  const bigint = parseInt(hex.replace('#', ''), 16);
  const r = (bigint >> 16) & 255;
  const g = (bigint >> 8) & 255;
  const b = bigint & 255;

  return `rgb(${r}, ${g}, ${b})`;
}
