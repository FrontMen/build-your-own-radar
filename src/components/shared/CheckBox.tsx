import React from 'react';
import styled from 'styled-components';
import { FaCheck } from 'react-icons/fa';
import { lightTheme } from 'src/Theme';
import { Typography } from 'src/Theme/Typography';

const CheckBoxContainer = styled.span<Pick<CheckBoxProps, 'checked'>>`
  border: 2px solid ${props => props.theme.pallet.primary};
  background-color: ${props =>
    props.theme.pallet[props.checked ? 'primary' : 'white']};
  /* border-radius: 4px; */
  padding: 2px;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  position: relative;
  width: 24px;
  height: 24px;
  ${Typography.body};
`;

const Label = styled.label`
  ${Typography.body};
  margin-right: ${props => props.theme.space[3]}px;
`;
const Container = styled.div`
  display: flex;
  justify-content: space-between;
  margin: ${props => props.theme.space[2]}px;
  /* width: 100%; */
  align-items: center;
`;

interface CheckBoxProps {
  checked: boolean;
  label: string;
  onClick: () => void;
}

export const CheckBox: React.FC<CheckBoxProps> = ({
  checked,
  label,
  onClick,
}) => {
  return (
    <Container>
      <Label>{label}</Label>
      <CheckBoxContainer
        data-testid="checkbox"
        checked={checked} onClick={onClick}>
        {checked && <FaCheck color={lightTheme.pallet.white} />}
      </CheckBoxContainer>
    </Container>
  );
};
