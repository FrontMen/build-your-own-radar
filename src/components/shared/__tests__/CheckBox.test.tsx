import React from 'react';
import { fireEvent } from '@testing-library/dom';

import { CheckBox } from '../CheckBox';
import { withAllProviders } from 'test/helpers';

const defaultProps = {
  checked: false,
  label: 'testCheckbox',
  onClick: jest.fn(),
};

describe('CheckBox', () => {
  it('should matches the snapshot when unchecked', () => {
    const { container } = withAllProviders(<CheckBox {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('should matches the snapshot when checked', () => {
    const { container } = withAllProviders(
      <CheckBox {...defaultProps} checked={true} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should call onClick callback when clicked', () => {
    const { getByTestId } = withAllProviders(<CheckBox {...defaultProps} />);

    fireEvent.click(getByTestId('checkbox'));
    expect(defaultProps.onClick).toHaveBeenCalled();
  });
});
