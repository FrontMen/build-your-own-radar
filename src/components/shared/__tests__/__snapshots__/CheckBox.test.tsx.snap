// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox should matches the snapshot when checked 1`] = `
.c2 {
  border: 2px solid #526de4;
  background-color: #526de4;
  padding: 2px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  position: relative;
  width: 18px;
  height: 18px;
  font-family: Roboto,san-serif;
  font-size: 1em;
  font-weight: 400;
  line-height: 1.65;
}

.c1 {
  font-family: Roboto,san-serif;
  font-size: 1em;
  font-weight: 400;
  line-height: 1.65;
  margin-right: 16px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
    >
      testCheckbox
    </label>
    <span
      class="c2"
      data-testid="checkbox"
    >
      <svg
        color="#fff"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(255, 255, 255);"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`CheckBox should matches the snapshot when unchecked 1`] = `
.c2 {
  border: 2px solid #526de4;
  background-color: #fff;
  padding: 2px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  position: relative;
  width: 18px;
  height: 18px;
  font-family: Roboto,san-serif;
  font-size: 1em;
  font-weight: 400;
  line-height: 1.65;
}

.c1 {
  font-family: Roboto,san-serif;
  font-size: 1em;
  font-weight: 400;
  line-height: 1.65;
  margin-right: 16px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
    >
      testCheckbox
    </label>
    <span
      class="c2"
      data-testid="checkbox"
    />
  </div>
</div>
`;
